### ISSUE TITLE
Manage Tags: default namespace option for new tags

### ISSUE THREAD

**Issue by novakda** [hydrusnetwork/hydrus#406]
When adding many different tags of the same namespace to different images, it can be time consuming to type the namespace each time, especially for tags that don't yet exist in the database.

Please add a dropdown to the manage tags dialog, or some similar option to specify a namespace to prepend to tags if none is specified.   

e.g. Set a default namespace "person".  In the manage tags dialog, add a "john smith" tag to a file (assuming there are no "john smith" tags in any namespace already).   The system recognizes no such tag exists, and displays "person:john smith" in the results, perhaps with some indication that this tag doesn't exist yet.  For consistency, it would be nice if it otherwise behaves the same as if the new tag did exist in the database.

**Comment by pozieuto** [comment]
So, in essence, when entering a new tag, you want the suggestions to be populated by user-defined namespaced versions, in addition to the default unnamespaced version? This seems like a useful addition of functionality, but it would definitely have to be an opt-in sort of thing, or users will be confused as to why there are suggestions for a tag that shouldn't already exist.

Suggestions of non-existent tags would already be indicated with a (0) at the end, presumably, but it would be easier to read if there were a dedicated designation for this. Perhaps prepending it with `* `?

**Comment by novakda** [comment]
> So, in essence, when entering a new tag, you want the suggestions to be populated by user-defined namespaced versions, in addition to the default unnamespaced version? This seems like a useful addition of functionality, but it would definitely have to be an opt-in sort of thing, or users will be confused as to why there are suggestions for a tag that shouldn't already exist.
> 
> Suggestions of non-existent tags would already be indicated with a (0) at the end, presumably, but it would be easier to read if there were a dedicated designation for this. Perhaps prepending it with `* `?

Yes, something like that, but I'm not sure what I've suggested would be the best UI. 

My use case is adding tags while reviewing images, all in the same namespace, but I don't know if they already exist in the database.  For example I might be filling in missing series or character tags.  It would be nice not to have to type 'series:' or 'character:' each time when adding a new one.

Perhaps a dedicated or floating "add tags" dialog with a dropdown for namespaces and a textbox that shows suggestions (defaulting to searching only what namespace is selected)? 

This could probably be done with an app using the API, but there's no way to get tags without getting files right now.

**Comment by DonaldTsang** [comment]
So on the fly tag creation while managing media? sounds good.

**Comment by ShadowJonathan** [comment]
Maybe prepend with `:` instead, so that the namespace is "empty", and hydrus fills in with most common/predetermined sets of namespaces in the suggestion list?

**Comment by novakda** [comment]
One way I envisioned this: add a dropdown above the tag entry textbox in 'manage tags' that lists all available namespaces + '(none selected)'.  If one is selected, autocomplete in the textbox only searches for tags in that namespace, as if you'd typed the namespace before you started typing.  If the tag does not exist, it adds it just like if you'd typed the whole thing in.

This allows adding tags in the same namespace more quickly when you're cleaning up a bunch of files.
